### PR TITLE
Make `run` available (but deprecated) in the embedding

### PR DESCRIPTION
**Pull request by sigurdm**

### CODE REVIEW

File: test/embedding/goldens/helptext.txt [hunk 1]
```diff
@@ -68,6 +69,7 @@ Available subcommands:
   outdated    Analyze your dependencies to find which ones can be upgraded.
   publish     Publish the current package to pub.dartlang.org.
   remove      Removes a dependency from the current package.
+  run         DEPRECATED: Use `dart run`.
```
Comment (jonasfj) [inline test/embedding/goldens/helptext.txt:72]: https://pub.dev/documentation/args/latest/command_runner/Command/hidden.html

File: lib/src/command/run.dart [hunk 2]
```diff
@@ -17,15 +17,19 @@ class RunCommand extends PubCommand {
...
+      : 'Run an executable from a package.';
   @override
   String get argumentsDescription => '<executable> [arguments...]';
   @override
   String get docUrl => 'https://dart.dev/tools/pub/cmd/pub-run';
   @override
   bool get allowTrailingOptions => false;
 
-  RunCommand() {
+  final bool deprecated;
```
Comment (jonasfj) [inline lib/src/command/run.dart:30]: Let's just set: https://pub.dev/documentation/args/latest/command_runner/Command/hidden.html

Comment (sigurdm) [reply lib/src/command/run.dart:32]: Done

File: test/embedding/goldens/helptext.txt (hunk 1 above)
Comment (sigurdm) [reply test/embedding/goldens/helptext.txt]: Done

**Review by jonasfj — APPROVED** [review]

**Comment by natebosch** [comment]
If I understand correctly this will warn if you use `dart pub run` and recommend instead `dart run`.

The latter does not solve all the same use cases. For example running a binary from another package like `pub run build_runner`. How do we anticipate users running for that use case?

**Comment by jonasfj** [comment]
@natebosch, we aim to make it so that `dart run <package>[:<command>]` will work the same way `pub run <package>[:<command>]`.

**Pull request merged**